com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolumePerSceneData.cs

```diff
@@ -126,19 +129,16 @@ void OnDestroy()
             QueueAssetRemoval();
         }
 
-        void Update()
+        public void SetBakingState(int state)
         {
-            // Query state from ProbeReferenceVolume.instance.
-            // This is temporary here until we implement a state system.
-            m_CurrentState = ProbeVolumeState.Default;
+            if (state == m_CurrentState)
+                return;
 
-            if (m_PreviousState != m_CurrentState)
-            {
-                if (assets.ContainsKey(m_PreviousState) && assets[m_PreviousState] != null)
-                    ProbeReferenceVolume.instance.AddPendingAssetRemoval(assets[m_PreviousState]);
+            if (assets.ContainsKey(m_CurrentState) && assets[m_CurrentState] != null)
+                ProbeReferenceVolume.instance.AddPendingAssetRemoval(assets[m_CurrentState]);
 
-                QueueAssetLoading();
-            }
+            m_CurrentState = state;
+            QueueAssetLoading();
```

Comment:
We def. don't want to queue an asset loading if the state is the same.

Reply:
there's a `if (state == m_CurrentState)` at the start of the function that early out in that case